dm/pkg/checker/table_structure.go

```diff
@@ -18,23 +18,41 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
+	"math"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/pingcap/errors"
 	column "github.com/pingcap/tidb-tools/pkg/column-mapping"
 	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb-tools/pkg/filter"
+	"github.com/pingcap/tidb/parser"
 	"github.com/pingcap/tidb/parser/ast"
 	"github.com/pingcap/tidb/parser/charset"
 	"github.com/pingcap/tidb/parser/model"
 	"github.com/pingcap/tidb/parser/mysql"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/dm/pkg/utils"
+)
+
+const (
+	// AutoIncrementKeyChecking is an identification for auto increment key checking.
+	AutoIncrementKeyChecking = "auto-increment key checking"
+	maxThreadNum             = 32
```

Comment:
dumpling use 4 as default value, how about set it to the same value.

Reply:
The only effect is that the time will be quadrupled. At the beginning, @sunzhaoyang  said he hopes the time can be controlled within 1 minute.

Reply:
Did you have a benchmark? For some customer, they would rather take a little longer than affect the upstream. And I think the time needed for checking is less than the time needed for dump data.

Reply:
test result is 

####  10000 tables
|added delay|spend time|
|-----------|---------|
|10 ms|9.5s|
|20 ms|15.6s|
|50 ms|35s|
|100 ms|68.5s|

I will discuss about upstream affact with PM further.